### ISSUE TITLE
What sort of security should storage Backends have?

### ISSUE THREAD

**Issue by joyously** [mavoweb/mavo#265]
https://github.com/mavoweb/mavo/blob/0579f87c474255b0530020a8584a939bc3f062ca/src/backend.github.js#L19

I am basing my code for a WordPress storage backend on a mix of this one and others and wondered about security. For GitHub, this check would allow a path of `.gitignore` or `.eslintrc.json` , but should it? 
For my code, I'm leery of `.htaccess` or `.htpasswd`. I already remove `../` regardless. 
I don't know if hidden folders should be a concern. 
I think I should limit the file extensions based on what WordPress already does (so they can't put PHP files on the site), but I'm not sure at what point it should fail (load, login, edit, save). That makes me think that maybe a PHP file extension should be read-only. But even images can have embedded hacks, so my concern is that I don't want my code to be the weak point that hackers exploit.

**Comment by joyously** [comment]
I haven't worked on this in a year, partly because WordPress by default does not allow json files. But I think I can still do it (if I work on it).

This article was mentioned in a WordPress dev chat today, and it got me wondering about json vulnerabilities in Mavo, especially from the viewpoint of a storage backend in PHP: 
https://medium.com/node-security/the-most-common-xss-vulnerability-in-react-js-applications-2bdffbcc1fa0
If the user enters data containing script tags and JS, can it get executed when retrieved from storage?

**Comment by LeaVerou** [comment]
If the user has no access to that file, they cannot use Mavo to view it or edit it, so I'm not sure how it can be a security risk?
I can certainly see use cases for using Mavo to edit `.eslintrc.json` and `.gitignore`!

**Comment by joyously** [comment]
> If the user enters data containing script tags and JS, can it get executed when retrieved from storage?

Like explained in the article, if the data contains a script tag and code, is it possible to be executed when retrieved?

**Comment by LeaVerou** [comment]
I don't believe so. There was a similar XSS flaw in the Markdown plugin when Mavo was first released, but it was fixed about a year ago.
Of course there's always the chance I could have misunderstood something! In that case, a demo demonstrating the vulnerability would work best :)
